csrc/multidevice/runtime.h

```diff
@@ -25,9 +25,7 @@ namespace nvfuser {
 class MultiDeviceRuntime {
  public:
   explicit MultiDeviceRuntime(Pipeline* pipeline, Communicator& comm)
-      : pipeline_(pipeline), comm_(comm) {
-    validate();
```

Comment:
Why is validation moved?

Reply:
Indeed, I moved the call to `validate` from the constructor to the method `runWithInput`. Why is it a problem? Why not do so? We can consider that It is always valid to virtually define the runtime, but possibly invalid to run it with concrete input.
This way, the user can instantiate the `MultiDeviceRuntime`, and then call `validate` to query whether the runtime is valid. This is done in the Gtests, where we want to skip the test rather than throwing an error if the runtime setup is not compatible with the Pipeline.

Reply:
OK, makes sense.